pyccel/codegen/printing/fcode.py

```diff
@@ -1189,14 +1212,16 @@ def _print_AliasAssign(self, expr):
         return self._get_statement(code) + '\n'
 
     def _print_CodeBlock(self, expr):
-        body = []
-        for b in expr.body:
+        body_exprs, new_vars = expand_to_loops(expr, self.parser.get_new_variable, language_has_vectors = True)
+        self.add_vars_to_namespace(*new_vars)
+        body_stmts = []
+        for b in body_exprs :
             line = self._print(b)
             if (self._additional_code):
-                body.append(self._additional_code)
+                body_stmts.append(self._additional_code)
                 self._additional_code = None
-            body.append(line)
-        return ''.join(body)
+            body_stmts.append(line)
+        return ''.join(self._print(b) for b in body_stmts)
```

Comment:
I've just realized that this function is a bit different than the one in the C printer:

1. In the C printer we set `self._additional_code = ''` (empty string), while here we set it to `None`
2. In the C printer the statements are missing the final newline character, so that has to be added when joining lines

Regarding point 1 I don't think we should store the additional code in a string, so the Fortran printer's behavior is preferable. Similarly about point 2, I think it's cleaner to have one-line statements end with `\n`. What do you think?

Reply:
I agree that point 2 needs to also be done for the C printer. We are starting to have empty lines in the C code more often so I think this should be fixed.
I'm not sure there is much difference for point 1. In both cases the additional code is stored in a string so that it is printed in the right order. The only difference is the value when it is empty

In any case I don't think these points should be fixed in this PR as it is already complicated enough on its own

Reply:
OK. Can you please create an issue for point 2?

Reply:
Done. See issue #761

Reply:
Thanks!